unidist/core/backends/mpi/core/controller/garbage_collector.py

```diff
@@ -63,8 +62,11 @@ def _send_cleanup_request(self, cleanup_list):
         # Cache serialized list of data IDs
         s_cleanup_list = SimpleDataSerializer().serialize_pickle(cleanup_list)
         async_operations = AsyncOperations.get_instance()
-        for rank_id in range(initial_worker_number, mpi_state.world_size):
-            if rank_id != mpi_state.rank:
+        for rank_id in range(mpi_state.global_size):
```

Comment:
Can we use mpi_state.workers?

Reply:
Yes, but both workers and monitors

Reply:
Let's use it to avoid multiple checks `not mpi_state.is_root_process(rank_id)` on every iteration.